api/models/comment-models.js

```diff
@@ -0,0 +1,14 @@
+const connection = require("../../db/connection");
+const { checkExists } = require("../../api/utility-functions/api-utils");
+
+exports.removeCommentById = (commentid) => {
+  const promiseCatcher = [checkExists("comments", "comment_id", commentid)];
+  const valueCatcher = [commentid];
+  let queryString = `DELETE FROM comments WHERE comment_id = $1`;
+
+  return Promise.all(promiseCatcher).then(() => {
+    return connection.query(queryString, valueCatcher).then(({ rowCount }) => {
+      return rowCount;
+    });
+  });
```

Comment:
It's a valid approach to use checkExists first as a way to first check if the given comment exists - but as you mentioned in your note, there's no need to use a Promise.all for this.

A promise.all is used to resolve multiple promises, but for just one you can use a .then on it directly. The promise.all doesn't actually do anything extra compared to just using a .then.

Note that you can also get around using checkExists by using the rowCount, inferring that if 0 rows were deleted then the comment didn't exist. (you could also use a RETURNING * and look directly at the rows that return from the deletion query)